controllers/flowlogspipeline/flp_service_monitor.go

```diff
@@ -0,0 +1,63 @@
+package flowlogspipeline
+
+import (
+	"context"
+
+	"github.com/netobserv/network-observability-operator/controllers/constants"
+	"github.com/netobserv/network-observability-operator/controllers/reconcilers"
+	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+func AddPrometheusServiceMonitor(ctx context.Context, b *builder, cl reconcilers.ClientHelper) error {
+	logger := log.FromContext(ctx)
+	svcKey := types.NamespacedName{
+		Name:      b.promServiceName(),
+		Namespace: b.namespace,
+	}
+	svc := v1.Service{}
+	err := cl.Client.Get(ctx, svcKey, &svc)
+	if errors.IsNotFound(err) {
+		logger.Info("flowlogs-pipeline prom service not found; not creating the service monitor")
+		return nil
+	}
+	serviceMonitorObject := buildPrometheusServiceMonitorObject(b)
+	// apply object to kubernetes
+	if err = cl.CreateOwned(ctx, serviceMonitorObject); err != nil {
+		return err
+	}
+	return nil
+}
+
+func buildPrometheusServiceMonitorObject(b *builder) *monitoringv1.ServiceMonitor {
```

Comment:
This function is very similar to `consolePluginServiceMonitor` --- Can you (1) make sure that at least the names will be similar ---starting with the word build and finishing with Object makes sense to me so s/consolePluginServiceMonitor/buildConsolePluginServiceMonitorObject 

Also, if you can re-use the same code in both functions by taking repeated code into a helper function it will be best.